### PR TITLE
#505 Apply conan conventions

### LINKED ISSUE #505: Please add gcc 8 builds for readline
### Description of Problem, Request, or Question
gcc 8 builds are missing
### Package Details (Include if Applicable)
* Package Name/Version: **readline/7.0**
* Operating System: **Fedora 28**
* Compiler+version: **gcc-8.2**

### LINKED ISSUE #507: readline needs link to termcap
### Description of Problem, Request, or Question
I am packaging an executable that needs readline.
Linking fails because of undefined references:
```
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(display.o): In function `_rl_move_cursor_relative':
display.c:(.text+0x10eb): undefined reference to `tputs'
display.c:(.text+0x1157): undefined reference to `tputs'
display.c:(.text+0x13bf): undefined reference to `tputs'
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(display.o): In function `_rl_move_vert':
display.c:(.text+0x14ec): undefined reference to `tputs'
display.c:(.text+0x1550): undefined reference to `tputs'
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(display.o):display.c:(.text+0x2318): more undefined references to `tputs' follow
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(display.o): In function `update_line':
display.c:(.text+0x3263): undefined reference to `tgoto'
display.c:(.text+0x3277): undefined reference to `tputs'
display.c:(.text+0x334d): undefined reference to `tputs'
display.c:(.text+0x3620): undefined reference to `tputs'
display.c:(.text+0x375b): undefined reference to `tgoto'
display.c:(.text+0x376d): undefined reference to `tputs'
display.c:(.text+0x3a5a): undefined reference to `tputs'
display.c:(.text+0x3ae4): undefined reference to `tputs'
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(display.o): In function `rl_redisplay':
display.c:(.text+0x55fa): undefined reference to `tputs'
display.c:(.text+0x57e5): undefined reference to `tputs'
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(display.o):display.c:(.text+0x59b8): more undefined references to `tputs' follow
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(terminal.o): In function `_rl_get_screen_size':
terminal.c:(.text+0xf9): undefined reference to `tgetnum'
terminal.c:(.text+0x13a): undefined reference to `tgetnum'
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
terminal.c:(.text+0x681): undefined reference to `PC'
terminal.c:(.text+0x695): undefined reference to `BC'
terminal.c:(.text+0x69f): undefined reference to `UP'
terminal.c:(.text+0x712): undefined reference to `tgetent'
terminal.c:(.text+0x747): undefined reference to `tgetstr'
terminal.c:(.text+0x775): undefined reference to `PC'
terminal.c:(.text+0x78d): undefined reference to `BC'
terminal.c:(.text+0x7a1): undefined reference to `UP'
terminal.c:(.text+0x7b6): undefined reference to `tgetflag'
terminal.c:(.text+0x814): undefined reference to `tgetflag'
terminal.c:(.text+0x9f8): undefined reference to `tgetflag'
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(terminal.o): In function `_rl_backspace':
terminal.c:(.text+0xcf0): undefined reference to `tputs'
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(terminal.o): In function `rl_ding':
terminal.c:(.text+0xda9): undefined reference to `tputs'
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(terminal.o): In function `_rl_enable_meta_key':
terminal.c:(.text+0xe17): undefined reference to `tputs'
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(terminal.o): In function `_rl_disable_meta_key':
terminal.c:(.text+0xe79): undefined reference to `tputs'
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(terminal.o): In function `_rl_control_keypad':
terminal.c:(.text+0xead): undefined reference to `tputs'
/home/maarten/.conan/data/readline/7.0/bincrafters/stable/package/1d877a3df840030e6a8abb74c5ffb9088d08b47a/lib/libreadline.a(terminal.o):terminal.c:(.text+0xf0b): more undefined references to `tputs' follow
collect2: error: ld returned 1 exit status
```
### Package Details (Include if Applicable)
* Package Name/Version: **readline/7.0**
* Operating System: **Fedora 28**
* Compiler+version: **gcc-8.2**

### Steps to reproduce (Include if Applicable)
Create an executable and link to readline, use the `update_line` function.

### How to fix
It appears readline is configured without ncurses, so termcap is used.
So readline should add `termcap` to `self.cpp_info.libs` in `package_info`.
(Should termcap be packaged?)

I manually added `termcap` to the linked libraries to use the termcap of my distribution.

### PR DESCRIPTION
**Pull request by uilianries**
closes https://github.com/bincrafters/community/issues/505
closes https://github.com/bincrafters/community/issues/507

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 linux: &linux
    os: linux
+   dist: xenial
    sudo: required
```
Comment (Croydon) [inline .travis.yml]: Please remove the sudo when applying the other changes. Travis is deprecating it.

Comment (uilianries) [reply .travis.yml]: @Croydon  is it no longer required for docker service?

Comment (Croydon) [reply .travis.yml]: If the docker service is getting used Travis is implicit setting sudo: true no matter what. And sudo disable environments are soon not available anymore. So let's remove it.

File: conanfile.py [hunk 2]
```diff
@@ -18,58 +17,54 @@ class ReadLineConan(ConanFile):
...
+            del self.options.shared
 
     def configure(self):
         del self.settings.compiler.libcxx
-        if self.settings.os == "Windows" and self.settings.compiler == "gcc":
-            logger.warn("Readline requires ncurses installed")
 
     def source(self):
         source_url = "https://git.savannah.gnu.org/cgit/readline.git/snapshot/readline"
         tools.get("{0}-{1}.tar.gz".format(source_url, self.version))
```
Comment (madebr) [inline conanfile.py]: Shouldn't this include a sha256?

**Comment by SSE4** [comment]
@uilianries  please fix appveyor
```
w64-mingw32/4.9.4/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -ltermcap
collect2.exe: error: ld returned 1 exit status
```

File: conanfile.py (hunk 2 above)
Comment (uilianries) [reply conanfile.py]: good idea

**Comment by uilianries** [comment]
@SSE4 I'm creating termcap package. It should be enough so solve that error.

**Comment by madebr** [comment]
@uilianries See bincrafters/community#507

File: conanfile.py [hunk 3]
```diff
@@ -69,5 +69,6 @@ def package(self):
 
     def package_info(self):
         self.cpp_info.libs = tools.collect_libs(self)
-        if self.settings.os == "Linux":
+        if self.settings.os == "Linux" or \
```
Comment (madebr) [inline conanfile.py]: Can't these lines be simplified to:
```suggestion
        if self.settings.compiler in ["gcc", "clang"]:
```

Comment (uilianries) [reply conanfile.py]: Yes, looks better, but is no longer required as termcap is a base requirement now.

**Comment by uilianries** [comment]
Finally! It's working! I need a coffee or maybe two! :coffee:

**Pull request merged**